src/plugins/map/MapObject.cpp

```diff
@@ -1080,25 +1099,20 @@ MapObjectMapLoc::~MapObjectMapLoc()
 
 void MapObjectMapLoc::UpdateFromParams(const MapLocParams& params)
 {
-	m_lineSize = params.lineSize;
-	m_width = params.width;
-	m_color = params.color;
-	m_circleRadius = params.circleRadius;
-	m_circleColor = params.circleColor;
-
-	Update(true);
+	m_mapLocParams = params;
 }
 
-MapLocParams MapObjectMapLoc::GetParams() const
+void MapLocTemplate::UpdateFromParams(const MapLocParams& params)
 {
-	MapLocParams params;
-	params.lineSize = m_lineSize;
-	params.width = m_width;
-	params.color = m_color;
-	params.circleRadius = m_circleRadius;
-	params.circleColor = m_circleColor;
+	m_mapLocParams = params;
+	m_MapLoc->UpdateFromParams(params);
 
-	return params;
+	m_MapLoc->Update(true);
+}
+
+MapLocParams* MapObjectMapLoc::GetParams()
+{
+	return &m_mapLocParams;
```

Comment:
this should return const-reference instead of by pointer. The only thing that uses it is dereferencing it anyways, so theres no point to returning a mutable pointer here.

Also make the function `const`